compiler/rustc_codegen_cranelift/src/driver/aot.rs

```diff
@@ -518,11 +518,5 @@ fn determine_cgu_reuse<'tcx>(tcx: TyCtxt<'tcx>, cgu: &CodegenUnit<'tcx>) -> CguR
     // know that later). If we are not doing LTO, there is only one optimized
     // version of each module, so we re-use that.
     let dep_node = cgu.codegen_dep_node(tcx);
-    assert!(
-        !tcx.dep_graph.dep_node_exists(&dep_node),
-        "CompileCodegenUnit dep-node for CGU `{}` already exists before marking.",
-        cgu.name()
-    );
```

Comment:
Can we add this assertion back in (if -Zincremental_verify_depnodes is active)? 
(assertion 1/3 🙂)